src/services/CertificateGenerationService.ts

```diff
@@ -177,6 +175,58 @@ class CertificateGenerationService {
             });
     }
 
+    /**
+     * Method to check if Welsh translation is enabled.
+     * @param featureFlags IFeatureFlags interface
+     * @returns boolean
+     */
+    public isGlobalWelshFlagEnabled(featureFlags: IFeatureFlags): boolean {
+        if (!featureFlags.welshTranslation.enabled) {
+            console.warn(`Unable to translate any test results: global Welsh flag disabled.`);
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Method to check if Welsh translation is enabled for the given test type.
+     * @param featureFlags IFeatureFlags interface
+     * @param testResult string of result, PASS/PRS/FAIL
+     * @returns boolean
+     */
+    public isTestResultFlagEnabled(featureFlags: IFeatureFlags, testResult: string): boolean {
+        let shouldTranslate: boolean = false;
+        switch (testResult) {
+            case TEST_RESULTS.PRS:
+                shouldTranslate = featureFlags.welshTranslation.translatePrsTestResult ? true : false;
```

Comment:
can you simplify these please
```suggestion
                shouldTranslate = featureFlags.welshTranslation.translatePrsTestResult;
```

Reply:
is it ever possible it wont be defined?

Reply:
In theory, I guess. This might be more appropriate for this then
```suggestion
                shouldTranslate = featureFlags.welshTranslation?.translatePrsTestResult;
```

Reply:
Is a good point though, we probably want to have some light validation on checking if there actually are feature flags coming back from the getProfile before it gets used

Reply:
I think with the above, if featureFlags or welshTranslation didn't exist it would return undefined.

```
featureFlags.welshTranslation?.translatePrsTestResult; 
```

Yes maybe some light validation to ensure it has retrieved the flag, otherwise default to false?

Reply:
Or, much like with Secrets Manager do we assume it is there (as it is being set up in the TF)?

Reply:
Fair point, have you got a test validating that it doesn't fall over if the above scenario did happen?

Reply:
No test for that, so I'll add those quickly for each Flag. I think Dan was right with making the flag optional. Something like this:
`public isGlobalWelshFlagEnabled(featureFlags?: IFeatureFlags): boolean {
        if (!featureFlags?.welshTranslation.enabled) {
            console.warn(`Unable to translate any test results: global Welsh flag disabled.`);
            return false;
        }
        return true;
    }`
    Testing that, it did return false. Same with the PASS/PRS/FAIL flags. Not sure if we need to make welshTranslation optional, as it'd either retrieve the whole thing or none?

Reply:
also doing
`shouldTranslate = featureFlags?.welshTranslation.translatePrsTestResult;`
resulted in shouldTranslate being of type `Boolean | undefined` so I think we need to keep
`shouldTranslate = featureFlags?.welshTranslation.translatePrsTestResult ? true : false;`